src/config/yaml.rs

```diff
@@ -108,19 +109,36 @@ impl YamlConfig {
         serde_yaml::from_reader(reader).map_err(|e| e.into())
     }
 
-    pub fn get(env: &EnvConfig) -> Result<Self> {
+    pub fn get(env: &EnvConfig) -> Result<YamlConfig> {
         if let Some(yaml) = env.config_yaml.as_ref() {
-            return Self::from_str(yaml);
+            // We're getting the configuration from the environment variable `NAVI_CONFIG_YAML`
+            let mut cfg = Self::from_str(yaml)?;
+            cfg.source = format!("NAVI_CONFIG_YAML={}", yaml);
```

Comment:
```suggestion
            cfg.source = format!("{}={}", env_var::CONFIG_YAML, yaml);
```
It is better to use `env_var::CONFIG_YAML` that is used for `env.config_yaml`.
This requires `use crate::env_var;`